### ISSUE TITLE
[Question] Decimal type in SummingMergeTree

### ISSUE THREAD

**Issue by igor-sh8** [yandex/ClickHouse#3282]
Hello. 

I tried to use decimal type in SummingMergeTree table (decimal column under aggregation). Insert and select queries works fine, but when I try to use OPTIMIZE TABLE statement for collapsing rows in the table I get next error: 
`DB::Exception: Method get64 is not supported for Decimal64`
I know that the decimal type is currently experimental, so that error is a bug, is it in development or maybe decimal type will not work with the SummingMergeTree type (clickhouse server version 18.12.17)?

Thanks in advance for the help.

**Comment by 4ertus2** [comment]
Hello,

Could you print here a sample the queries to reproduce the issue?

**Comment by igor-sh8** [comment]
Create table query:
`create table TestDec (Date Date, Id Int32, Sum Decimal64(8)) Engine=SummingMergeTree(Date,(Id),8192,(Sum))`

Optimize query: 
`optimize table TestDec`

Stack Trace: 

```
2018.10.03 15:35:50.275602 [ 14 ] <Error> void DB::BackgroundProcessingPool::threadFunction(): Code: 48, e.displayText() = DB::Exception: Method get64 is not supported for Decimal64, e.what() = DB::Exception, Stack trace:

0. /usr/bin/clickhouse-server(StackTrace::StackTrace()+0x16) [0x9491e46]
1. /usr/bin/clickhouse-server(DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x22) [0x3019952]
2. /usr/bin/clickhouse-server(DB::IColumn::get64(unsigned long) const+0x66) [0x3430ed6]
3. /usr/bin/clickhouse-server(DB::SummingSortedBlockInputStream::insertCurrentRowIfNeeded(std::vector<COWPtr<DB::IColumn>::mutable_ptr<DB::IColumn>, std::allocator<COWPtr<DB::IColumn>::mutable_ptr<DB::IColumn> > $
4. /usr/bin/clickhouse-server(DB::SummingSortedBlockInputStream::merge(std::vector<COWPtr<DB::IColumn>::mutable_ptr<DB::IColumn>, std::allocator<COWPtr<DB::IColumn>::mutable_ptr<DB::IColumn> > >&, std::priority_q$
5. /usr/bin/clickhouse-server(DB::SummingSortedBlockInputStream::readImpl()+0x209) [0x75098c9]
6. /usr/bin/clickhouse-server(DB::IProfilingBlockInputStream::read()+0x25a) [0x698ecfa]
7. /usr/bin/clickhouse-server(DB::MergeTreeDataMergerMutator::mergePartsToTemporaryPart(DB::MergeTreeDataMergerMutator::FuturePart const&, DB::MergeListEntry&, long, DB::DiskSpaceMonitor::Reservation*, bool)+0x2b$
8. /usr/bin/clickhouse-server(DB::StorageMergeTree::merge(bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool, std::__cxx11::basic_string<char, std::char_trait$
9. /usr/bin/clickhouse-server(DB::StorageMergeTree::backgroundTask()+0x135) [0x728c985]
10. /usr/bin/clickhouse-server(DB::BackgroundProcessingPool::threadFunction()+0x43e) [0x72fd4be]
11. /usr/bin/clickhouse-server() [0x9e659ef]
12. /lib/x86_64-linux-gnu/libpthread.so.0(+0x8064) [0x7f1985dc6064]
13. /lib/x86_64-linux-gnu/libc.so.6(clone+0x6d) [0x7f19853ee62d]
```

**Comment by 4ertus2** [comment]
It would be fixed for Decimal32 and Decimal64 in #3329
Decimal128 needs much more work to support. So there would be the same exception for it.

**Comment by 4ertus2** [comment]
I was wrong. It's not much work to support Decimal128 too #3348
